### PR TITLE
Privacy RPC multi-tenancy token public enclave key validation

### PR DESCRIPTION
**Pull request by jframe**
## PR description
Validates all privacy RPCs so that when using privacy multi-tenancy that the JWT/Vertx User has a enclavePublicKey.

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/MultiTenancyUserUtil.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.privacy.methods;
+
+import java.util.Optional;
+
+import io.vertx.ext.auth.User;
+
+public class MultiTenancyUserUtil {
+  private static final String PRIVACY_PUBLIC_KEY = "privacyPublicKey";
```
Comment (rain-on) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/MultiTenancyUserUtil.java]: what does this represent? I.e. guess its a fieldname in a principal right? Do we want to name it as such - eg enclavePublicKeyFieldName?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/MultiTenancyValidatingPrivacyRpcMethod.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.JsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+
+import java.util.Optional;
+
+import io.vertx.ext.auth.User;
+
+public class MultiTenancyValidatingPrivacyRpcMethod implements JsonRpcMethod {
```
Comment (rain-on) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/MultiTenancyValidatingPrivacyRpcMethod.java]: YMMV: add Decorator to classname?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/JsonRpcError.java [hunk 3]
```diff
@@ -115,6 +115,10 @@
   DECODE_ERROR(-50100, "Unable to decode the private signed raw transaction"),
   GET_PRIVATE_TRANSACTION_NONCE_ERROR(-50100, "Unable to determine nonce for account in group."),
 
+  // Privacy multi-tenancy errors
+  PRIVACY_MULTI_TENANCY_NO_TOKEN(-50100, "No token provided"),
```
Comment (rain-on) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/JsonRpcError.java]: nit: Is this actually just a http "unauthorised" response? Or is that too vague?
If this is the right way to go ... is "token" the right word" Or do we want to say something else?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/MultiTenancyValidatingPrivacyRpcMethod.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    final Optional<User> user = requestContext.getUser();
+    final Object id = requestContext.getRequest().getId();
+    if (user.isEmpty()) {
+      return new JsonRpcErrorResponse(id, PRIVACY_MULTI_TENANCY_NO_TOKEN);
+    } else if (MultiTenancyUserUtil.enclavePublicKey(user).isEmpty()) {
+      return new JsonRpcErrorResponse(id, PRIVACY_MULTI_TENANCY_NO_ENCLAVE_PUBLIC_KEY);
```
Comment (rain-on) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/MultiTenancyValidatingPrivacyRpcMethod.java]: probably need some indepth logging here "Supplied user token does not contain an Enclave Public Key. Confirm configuration of token-generator"? 
Probably need a good log msg for the above conditional too (i.e. Http request did not contain a JWT)

Comment (jframe) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/MultiTenancyValidatingPrivacyRpcMethod.java]: done

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/JsonRpcError.java (hunk 3 above)
Comment (jframe) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/JsonRpcError.java]: done

**Pull request merged**